google-ads-examples/src/main/java/com/google/ads/googleads/examples/planning/ForecastReach.java

```diff
@@ -0,0 +1,352 @@
+
+    if (!params.parseArguments(args)) {
+
+      // Either pass the required parameters for this example on the command line, or insert them
+      // into the code here. See the parameter class definition above for descriptions.
+      params.customerId = Long.parseLong("INSERT_CUSTOMER_ID_HERE");
+    }
+
+    GoogleAdsClient googleAdsClient;
+    try {
+      googleAdsClient = GoogleAdsClient.newBuilder().fromPropertiesFile().build();
+    } catch (FileNotFoundException fnfe) {
+      System.err.printf(
+          "Failed to load GoogleAdsClient configuration from file. Exception: %s%n", fnfe);
+      return;
+    } catch (IOException ioe) {
+      System.err.printf("Failed to create GoogleAdsClient. Exception: %s%n", ioe);
+      return;
+    }
+
+    try {
+      new ForecastReach().runExample(googleAdsClient, params.customerId);
+    } catch (GoogleAdsException gae) {
+      // GoogleAdsException is the base class for most exceptions thrown by an API request.
+      // Instances of this exception have a message and a GoogleAdsFailure that contains a
+      // collection of GoogleAdsErrors that indicate the underlying causes of the
+      // GoogleAdsException.
+      System.err.printf(
+          "Request ID %s failed due to GoogleAdsException. Underlying errors:%n",
+          gae.getRequestId());
+      int i = 0;
+      for (GoogleAdsError googleAdsError : gae.getGoogleAdsFailure().getErrorsList()) {
+        System.err.printf("  Error %d: %s%n", i++, googleAdsError);
+      }
+    }
+  }
+
+  /**
+   * Runs the example.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the customer ID for the reach forecast.
+   */
+  private void runExample(GoogleAdsClient googleAdsClient, long customerId) {
+    String locationId = "2840"; // US
+    String currencyCode = "USD";
+    long budgetMicros = 5_000_000L; // $5 USD
+
+    try (ReachPlanServiceClient reachPlanServiceClient =
+        googleAdsClient.getLatestVersion().createReachPlanServiceClient()) {
+      showPlannableLocations(reachPlanServiceClient);
+      showPlannableProducts(reachPlanServiceClient, locationId);
+      forecastManualMix(reachPlanServiceClient, customerId, locationId, currencyCode, budgetMicros);
+      forecastSuggestedMix(
+          reachPlanServiceClient, customerId, locationId, currencyCode, budgetMicros);
+    }
+  }
+
+  /**
+   * Maps friendly names of plannable locations to location IDs usable with ReachPlanServiceClient.
+   *
+   * @param reachPlanServiceClient instance of Reach Plan Service client.
+   */
+  private void showPlannableLocations(ReachPlanServiceClient reachPlanServiceClient) {
+    ListPlannableLocationsRequest request = ListPlannableLocationsRequest.newBuilder().build();
+    ListPlannableLocationsResponse response =
+        reachPlanServiceClient.listPlannableLocations(request);
+
+    System.out.println("Plannable Locations:");
+    System.out.println("Name,\tId,\t,ParentCountryId");
```

Comment:
It looks like the formatting doesn't match between this line and the one below. This line has two tabs, the following line has one (so this won't layout correctly on all terminals). I'd suggest printing this in a human readable format rather than a synthetic CSV (e.g. "Plannable location with name '%s', ID '%d' and parent country ID '%s' found"). Alternatively, I would drop the tabs and print it in plain CSV format.